### PR TITLE
Automate Web Driver Testing for Pushes to Production

### PR DESCRIPTION
**Pull request by xwsxethan**
This change is [Reviewable](https://reviewable.io/reviews/uproxy/ufo-management-server-flask/136)

### CODE REVIEW

File: web_driver.sh [hunk 1]
```diff
@@ -57,6 +58,26 @@ function runUITests ()
   runInTestDirAndAssertCmd "python ui_test_suite.py --server_url='$SERVER_URL' --username='$USERNAME' --password='$PASSWORD'"
 }
 
+function runRemoteUITests ()
```
Comment (blueandgold) [inline web_driver.sh]: nit: I am not sure what is meant by "Remote" in the function name?  Would this rather be a local server in the travis environment?  If so, maybe `runFunctionalUITests()` might be better naming?

**Comment by xwsxethan** [comment]
Reviewed 4 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[web_driver.sh, line 61 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/136#-KH1qKGuLcaFLc5rXwD7-r1-61:-KHLmspXYZW6e4v4a3rj:65555836) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/383929c86d452ba36b2eee42ba387c1dd29bb536/web_driver.sh#L61)):*
<details><summary><i>Previously, blueandgold (Henry Chang) wrote…</i></summary>
> nit: I am not sure what is meant by "Remote" in the function name?  Would this rather be a local server in the travis environment?  If so, maybe `runFunctionalUITests()` might be better naming?
> 
> 
> 
> Or does this mean it's actually testing against ufo-prod?  Maybe add a comment to describe what server the tests are running against?</details>
Your second guess is more accurate. I've renamed them now to be more reflective of what is happening plus added comments.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/136)*

**Comment by blueandgold** [comment]
:lgtm: 

<details><summary><i>Previously, xwsxethan (Ethan) wrote…</i></summary>
> #### Automate Web Driver Testing for Pushes to Production
> 
> 
> 
> 
> </details>


---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[web_driver.sh, line 61 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/136#-KH1qKGuLcaFLc5rXwD7-r1-61:-KHM1oZsqIfZpnqJRIDx:-1236444263) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/383929c86d452ba36b2eee42ba387c1dd29bb536/web_driver.sh#L61)):*
<details><summary><i>Previously, xwsxethan (Ethan) wrote…</i></summary>
> Your second guess is more accurate. I've renamed them now to be more reflective of what is happening plus added comments.</details>
Ah, got it now.  So, sorry to go back on this, but your previous runUITests() method naming above is great, and if we can, let's go back to that?  Because it can either be used on either local or remote servers.

For this method, I think a tweak to the naming would be awesome, such as:
`runUITestsViaSauceLabs()` or `runUITestsWithSauceLabs()` instead of 'On SauceLabs'

Also, it would be nice to add a comment to describe why & how we are using SauceLabs.  If this is already documented elsewhere, then a link to that documentation would be awesome.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/136#-:-KHM2uUc6EQltuRp-_wg:980077423)*

**Comment by xwsxethan** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[web_driver.sh, line 61 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/136#-KH1qKGuLcaFLc5rXwD7-r1-61:-KHM8Gv-DI9IaCETlx-z:-499146009) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/383929c86d452ba36b2eee42ba387c1dd29bb536/web_driver.sh#L61)):*
<details><summary><i>Previously, blueandgold (Henry Chang) wrote…</i></summary>
> Ah, got it now.  So, sorry to go back on this, but your previous runUITests() method naming above is great, and if we can, let's go back to that?  Because it can either be used on either local or remote servers.
> 
> For this method, I think a tweak to the naming would be awesome, such as:
> `runUITestsViaSauceLabs()` or `runUITestsWithSauceLabs()` instead of 'On SauceLabs'
> 
> Also, it would be nice to add a comment to describe why & how we are using SauceLabs.  If this is already documented elsewhere, then a link to that documentation would be awesome.</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/136)*

**Pull request merged**